pydarn/plotting/power.py

```diff
@@ -0,0 +1,208 @@
+            date = records_of_interest[0]['origin.time']
+            stid = records_of_interest[0]['stid']
+            radar_abbrev = pydarn.SuperDARNRadars.radars[stid].\
+                hardware_info.abbrev
+
+            for record in records_of_interest:
+
+                # check which statistical parameter is to be plotted and
+                # then compute that parameter and update the record to
+                # include an additional field called 'noise'
+                if parameter == 'mean':
+                    mean_pwr0 = np.mean(record['pwr0'])
+                    record.update({'noise': mean_pwr0})
+                if parameter == 'median':
+                    median_pwr0 = np.median(record['pwr0'])
+                    record.update({'noise': median_pwr0})
+                if parameter == 'std':
+                    std = np.std(record['pwr0'])
+                    record.update({'noise': std})
+                if parameter == 'max':
+                    mx = max(record['pwr0'])
+                    record.update({'noise': mx})
+                if parameter == 'min':
+                    mn = min(record['pwr0'])
+                    record.update({'noise': mn})
+
+            plt.figure()
+            # use the time series RTP function
+            pydarn.RTP.plot_time_series(records_of_interest, parameter='noise',
+                                        beam_num=beam_num)
+            plt.ylabel(str(parameter).capitalize() + ' Power \n [raw units]')
+            plt.title(parameter.capitalize() + ' Lag 0 Power at ' +
+                      str(radar_abbrev) + ' on ' + str(
+                date) + '\n Beam: ' + str(
+                beam_num) + '  Frequency: ' + str(frequency) + ' kHz')
+            plt.show()
+
+        if compare is True:
+            # now compare frequencies separated by a frequency cutoff
+            # predefine the lists of separated records
+            low_freq_records = []
+            high_freq_records = []
+
+            # separate the records by frequency
+            for record in records:
+                if record['tfreq'] > frequency:
+                    high_freq_records.append(record)
+                if record['tfreq'] < frequency:
+                    low_freq_records.append(record)
+
+            # gather important info regarding the records of interest
+            # get the date information from the first record
+            date = low_freq_records[0]['origin.time']
+            # get the site location
+            stid = high_freq_records[0]['stid']
+            # site location in terms of abbreviation
+            radar_abbrev = pydarn.SuperDARNRadars.radars[
+                stid].hardware_info.abbrev
+
+            # check the length of the low and high frequency records to
+            # ensure there is actually data to plot
+            if len(low_freq_records) == 0 and len(high_freq_records) == 0:
+                print(
```

Comment:
@Cooper10 can you convert this to an exception, check out `rtp.py` and how it handles no data and check out /exceptions folder in pyDARN for how to make your own exceptions for the code.